serverlessrepo/parser.py

```diff
@@ -122,3 +124,23 @@ def parse_application_id(text):
     """
     result = re.search(APPLICATION_ID_PATTERN, text)
     return result.group(0) if result else None
+
+
+def strip_app_metadata(template_dict):
+    """
+    Strip the "AWS::ServerlessRepo::Application" metadata section from template.
+
+    :param template_dict: SAM template as a dictionary
+    :type template_dict: dict
+    :return: stripped template content
+    :rtype: str
+    """
+    template_dict_copy = copy.deepcopy(template_dict)
+
+    # strip the whole metadata section if SERVERLESS_REPO_APPLICATION is the only key in it
+    if not [k for k in template_dict_copy.get(METADATA) if k != SERVERLESS_REPO_APPLICATION]:
```

Comment:
This function seems to be assuming `get_app_metadata()` has already been called before calling this function, because it's not doing any validation that the template actually has metadata. This is brittle when looking at parser as a standalone module. You can (1) document the assumptions of the function or (2) make the function resilient to the case where there is no metadata section in the template. I prefer the latter. If there's no template metadata, it should return the same template back again.